File: maioridade.c
#include <stdio.h>

int main()
{
    int idade;
    while (1)
    {
        printf("Digite a idade (-1 para parar):\n");
        scanf("%d", &idade);
        if (idade == -1)
        {
            printf("Ok, obrigado por utilizar nosso programa.");
            break;
        }
        else if (idade >= 18)
        {
            printf("Maior de idade.\n");
        }
        else
        {
            printf("Menor de idade.\n");
        }
    }
    return 0;
}

File: Maioridade.java
import java.util.Scanner;

public class Maioridade {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int idade;

        System.out.println("Digite uma idade:");
        idade = scanner.nextInt();
        scanner.close();
        if (idade < 0) {
            System.out.println("Digite uma idade válida!");
        } else if (idade >= 18) {
            System.out.println(idade + " é maior de idade.");
        } else {
            System.out.println(idade + " é menor de idade.");
        }
    }
}
